### PR TITLE
sdk: Snapshot Create, Restore, and Enumerate

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:
sdk SnapshotCreate, SnapshotRestore, and SnapshotEnumerate calls

**Which issue(s) this PR fixes** (optional)
Part of #407

### CODE REVIEW

File: api/server/sdk/volume_snapshot.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+	"github.com/libopenstorage/openstorage/api"
+)
+
+// SnapshotCreate creates a read-only snapshot of a volume
+func (s *VolumeServer) SnapshotCreate(
+	ctx context.Context,
+	req *api.VolumeSnapshotCreateRequest,
+) (*api.VolumeSnapshotCreateResponse, error) {
+
```
Comment (sdeoras) [inline api/server/sdk/volume_snapshot.go]: ignoring input ctx? i would recommend checking if input ctx deadline has been requested by the client and take any appropriate action.

File: api/server/sdk/volume_snapshot.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		SnapshotId: snapshotID,
+	}, nil
+}
+
+// SnapshotRestore restores a volume to the specified snapshot id
+func (s *VolumeServer) SnapshotRestore(
+	ctx context.Context,
+	req *api.VolumeSnapshotRestoreRequest,
+) (*api.VolumeSnapshotRestoreResponse, error) {
+
```
Comment (sdeoras) [inline api/server/sdk/volume_snapshot.go]: same ctx comment as above

File: api/server/sdk/volume_snapshot.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	return &api.VolumeSnapshotRestoreResponse{}, nil
+}
+
+// SnapshotEnumerate returns a list of snapshots for the specified volume
+func (s *VolumeServer) SnapshotEnumerate(
+	ctx context.Context,
+	req *api.VolumeSnapshotEnumerateRequest,
+) (*api.VolumeSnapshotEnumerateResponse, error) {
+
```
Comment (sdeoras) [inline api/server/sdk/volume_snapshot.go]: same ctx comment as above

File: api/client/sdk/js/api_grpc_pb.js [hunk 4]
```diff
@@ -226,6 +226,72 @@ function deserialize_openstorage_api_VolumeInspectResponse(buffer_arg) {
...
+  if (!(arg instanceof api_pb.VolumeSnapshotRestoreResponse)) {
+    throw new Error('Expected argument of type openstorage.api.VolumeSnapshotRestoreResponse');
+  }
+  return new Buffer(arg.serializeBinary());
+}
+
+function deserialize_openstorage_api_VolumeSnapshotRestoreResponse(buffer_arg) {
+  return api_pb.VolumeSnapshotRestoreResponse.deserializeBinary(new Uint8Array(buffer_arg));
+}
+
```
[the remark this answers is not in the document]
Comment (lpabon) [reply api/client/sdk/js/api_grpc_pb.js]: yep

File: api/server/sdk/volume_snapshot.go (hunk 1 above)
Comment (lpabon) [reply api/server/sdk/volume_snapshot.go]: Yeah, we are going to do that for all grpc calls (CSI ones also) at a future change.

**Comment by lpabon** [comment]
Thanks for the review!

**Pull request merged**